world/perception/src/nodelets/aruco_detection.cpp

```diff
@@ -0,0 +1,131 @@
+#include <stdio.h>
+
+#include <ros/ros.h>
+#include <pluginlib/class_list_macros.hpp>
+#include <nodelet/nodelet.h>
+#include <std_msgs/Float64.h>
+#include <sensor_msgs/Image.h>
+
+#include <opencv2/core/hal/interface.h>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/opencv.hpp>
+#include <opencv2/videoio.hpp>
+
+#include <cv_bridge/cv_bridge.h>
+#include <interface/utils.hpp>
+#include <interface/stamped_markers.h>
+#include <aruco/aruco_nano.h>
+
+namespace perception
+{
+class aruco_detection_nodelet_t : public nodelet::Nodelet
+{
+public:
+  aruco_detection_nodelet_t()
+    : _publish_markers_img(false)
+    , _rgb_topic_name("/camera/rgb")
+    , _img_topic_name("/camera/markers")
+    , _markers_topic_name("/markers")
+    , _header(){};
+
+private:
+  virtual void onInit()
+  {
+    ros::NodeHandle& private_nh{ getPrivateNodeHandle() };
+    _rgb_topic_name = ros::this_node::getNamespace() + _rgb_topic_name;
+    _img_topic_name = ros::this_node::getNamespace() + _img_topic_name;
+    _markers_topic_name = ros::this_node::getNamespace() + _markers_topic_name;
+
+    bool publish_markers_img;
+    NODELET_PARAM_SETUP(private_nh, publish_markers_img);
+    _publish_markers_img = publish_markers_img;
+
+    _header.seq = 0;
+    _header.stamp = ros::Time::now();
+    _header.frame_id = _img_topic_name;
+
+    _markers_msg.header.seq = 0;
+    _markers_msg.header.stamp = ros::Time::now();
+    _markers_msg.header.frame_id = _markers_topic_name;
+
+    _frame_marker_publisher = private_nh.advertise<sensor_msgs::Image>(_img_topic_name, 1);
```

Comment:
Given that this is a sensor, I would prefer to have it as 1. If the network lags, we should focus on the most recent observation rather than old ones.